src/gui/updater/CMakeLists.txt

```diff
@@ -6,3 +6,53 @@ if(APPLE AND NOT BUILD_LIBRARIES_ONLY)
         set(OWNCLOUD_BUNDLED_RESOURCES ${OWNCLOUD_BUNDLED_RESOURCES} PARENT_SCOPE)
     endif()
 endif()
+
+target_sources(owncloudCore PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/ocupdater.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/ocupdater.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/updateinfo.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/updateinfo.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/updater.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/updater.h
+)
+
+if(SPARKLE_FOUND)
+    # Define this, we need to check in updater.cpp
+    target_compile_definitions(owncloudCore PUBLIC HAVE_SPARKLE)
+    target_sources(owncloudCore PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/sparkleupdater_mac.mm
+        ${CMAKE_CURRENT_SOURCE_DIR}/sparkleupdater.h
+    )
+    target_link_libraries(owncloudCore PRIVATE ${SPARKLE_LIBRARY})
+endif()
+
+if(WITH_APPIMAGEUPDATER)
+    # needed for set_source_files_properties(... TARGET_DIRECTORY ...)
+    cmake_minimum_required(VERSION 3.18)
+
+    message(STATUS "Including built-in libappimageupdate based updater")
+
+    set(appimageupdater_sources
+        ${CMAKE_CURRENT_SOURCE_DIR}/appimageupdateavailabledialog.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/appimageupdateavailabledialog.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/appimageupdateavailabledialog.ui
+        ${CMAKE_CURRENT_SOURCE_DIR}/appimageupdater.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/appimageupdater.h
+    )
+
+    target_sources(owncloudCore PRIVATE ${appimageupdater_sources})
+
+    # as libappimageupdate uses exceptions, we define a shim that catches all incoming exceptions
+    # and handles them in a suitable way
+    # we have to enable exceptions for this one compilation unit
+    set_source_files_properties(${appimageupdater_sources}
+        TARGET_DIRECTORY owncloudCore
+        PROPERTIES COMPILE_OPTIONS "-fexceptions"
+    )
+
+    target_compile_definitions(owncloudCore PRIVATE WITH_APPIMAGEUPDATER)
+    target_link_libraries(owncloudCore PRIVATE libappimageupdate-qt)
+
+    find_package(Threads REQUIRED)
```

Comment:
https://github.com/owncloud/client/blob/master/src/gui/CMakeLists.txt#L203

Reply:
Only enabled when the crashreporter is active, too, which might not be the case. Do you want to suggest to merge all the checks and include the dependency in a central place? Linking the dependency into the binaries twice shouldn't hurt either, though.

Reply:
I guess we can unconditionally depend on threads.